.github/scripts/trymerge_explainer.py

```diff
@@ -0,0 +1,123 @@
+import os
+import re
+from typing import List, Pattern, Tuple, Optional
+
+from trymerge import BOT_COMMANDS_WIKI, GitHubPR, gh_post_pr_comment
+
+CIFLOW_LABEL = re.compile(r"^ciflow/.+")
+CIFLOW_TRUNK_LABEL = re.compile(r"^ciflow/trunk")
+
+
+def has_label(labels: List[str], pattern: Pattern[str] = CIFLOW_LABEL) -> bool:
+    return len(list(filter(pattern.match, labels))) > 0
+
+
+class TryMergeExplainer(object):
+    force: bool
+    on_green: bool
+    land_checks: bool
+    pr: GitHubPR
+    org: str
+    project: str
+
+    has_trunk_label: bool
+    has_ciflow_label: bool
+
+    def __init__(
+        self,
+        force: bool,
+        on_green: bool,
+        land_checks: bool,
+        pr: GitHubPR,
+        org: str,
+        project: str,
+    ):
+        self.force = force
+        self.on_green = on_green
+        self.land_checks = land_checks
+        self.pr = pr
+        self.org = org
+        self.project = project
+
+    def get_flags(self) -> Tuple[bool, bool]:
+        self.has_trunk_label = has_label(self.pr.get_labels(), CIFLOW_TRUNK_LABEL)
+        self.has_ciflow_label = has_label(self.pr.get_labels(), CIFLOW_LABEL)
+        should_check_land_branch = self.land_checks and not self.has_trunk_label
+        should_check_green = self.on_green or self.has_ciflow_label
+
+        return (should_check_green, should_check_land_branch)
+
+    def _get_flag_msg(self) -> str:
+        if self.force:
+            return " the force (-f)"
+        elif self.on_green:
+            return " the green (-g)"
+        elif self.land_checks:
+            return " the land check (-l)"
+        else:
+            return "out a "
+
+    def _get_land_check_progress(self, commit: Optional[str]) -> str:
+        if commit is not None:
+            return (
+                f" and [land check]({BOT_COMMANDS_WIKI}) "
+                + f"progress [here](https://hud.pytorch.org/{self.org}/{self.project}/commit/{commit})"
+            )
+        else:
+            return ""
+
+    def _get_flag_explaination_message(self) -> str:
+        if self.force:
+            return (
+                "This means your PR will be merged immediately, bypassing any checks."
+            )
+        elif self.on_green:
+            return "This means that your PR will be merged once all signals have passed."
+        elif self.land_checks:
+            if self.has_trunk_label:
+                land_check_msg_suffix = (
+                    f"have run since you have the {CIFLOW_TRUNK_LABEL}."
+                )
+            else:
+                land_check_msg_suffix = (
+                    "and the land checks branch have run (ETA 4 Hours)."
```

Comment:
```suggestion
                    "and the land checks have run (ETA 4 Hours)."
```
Do we have access to the land checks link here? Can we tell people to track their progress there too?

Reply:
Maybe that will get rolled up into the comment already, not sure what the whole comment looks like for each--is there a place I can see the combined versions?